x/gamm/pool-models/balancer/pool_test.go

```diff
@@ -33,131 +33,97 @@ var (
-					uosmoValueOriginal = 1_000_000_000_000
-					atomValueOriginal  = 123
-					ionValueOriginal   = 657
-
-					// Weight does not affect calculations so it is shared
-					weight = 100
-				)
-
-				newLiquidity := sdk.NewCoins()
-
-				originalPoolAssetsByDenom := map[string]balancer.PoolAsset{
-					"uosmo": {
-						Token:  sdk.NewInt64Coin("uosmo", uosmoValueOriginal),
-						Weight: sdk.NewInt(weight),
-					},
-					"atom": {
-						Token:  sdk.NewInt64Coin("atom", atomValueOriginal),
-						Weight: sdk.NewInt(weight),
-					},
-					"ion": {
-						Token:  sdk.NewInt64Coin("ion", ionValueOriginal),
-						Weight: sdk.NewInt(weight),
-					},
-				}
-
-				return newLiquidity, originalPoolAssetsByDenom, originalPoolAssetsByDenom
+			name:         "new liquidity has no coins",
+			newLiquidity: sdk.NewCoins(),
+			originalPoolAssets: map[string]balancer.PoolAsset{
+				"uosmo": {
+					Token:  sdk.NewInt64Coin("uosmo", uosmoValueOriginal),
+					Weight: sdk.NewInt(weight),
+				},
+				"atom": {
+					Token:  sdk.NewInt64Coin("atom", atomValueOriginal),
+					Weight: sdk.NewInt(weight),
+				},
+				"ion": {
+					Token:  sdk.NewInt64Coin("ion", ionValueOriginal),
+					Weight: sdk.NewInt(weight),
+				},
 			},
+			expectPass: true,
 		},
 		{
 			name: "newLiquidity has a coin that poolAssets don't",
-
-			setup: func() (sdk.Coins, map[string]balancer.PoolAsset, map[string]balancer.PoolAsset) {
-				const (
-					uosmoValueOriginal = 1_000_000_000_000
-
-					// Weight does not affect calculations so it is shared
-					weight = 100
-				)
-
-				newLiquidity := sdk.NewCoins(
-					sdk.NewInt64Coin("juno", 1_000))
-
-				originalPoolAssetsByDenom := map[string]balancer.PoolAsset{
-					"uosmo": {
-						Token:  sdk.NewInt64Coin("uosmo", uosmoValueOriginal),
-						Weight: sdk.NewInt(weight),
-					},
-				}
-
-				return newLiquidity, originalPoolAssetsByDenom, originalPoolAssetsByDenom
+			newLiquidity: sdk.NewCoins(
+				sdk.NewInt64Coin("juno", 1_000)),
+			originalPoolAssets: map[string]balancer.PoolAsset{
+				"uosmo": {
+					Token:  sdk.NewInt64Coin("uosmo", uosmoValueOriginal),
+					Weight: sdk.NewInt(weight),
+				},
 			},
-
-			err: fmt.Errorf(balancer.ErrMsgFormatFailedInterimLiquidityUpdate, "juno"),
+			expectPass: false,
+			err:        fmt.Errorf(balancer.ErrMsgFormatFailedInterimLiquidityUpdate, "juno"),
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			newLiquidity, originalPoolAssetsByDenom, expectedPoolAssetsByDenom := tc.setup()
-
-			err := balancer.UpdateIntermediaryPoolAssetsLiquidity(newLiquidity, originalPoolAssetsByDenom)
+			expectedPoolAssetsByDenom := map[string]balancer.PoolAsset{}
+			for i, asset := range tc.originalPoolAssets {
+				expectedValue := balancer.PoolAsset{Token: asset.Token, Weight: asset.Weight}
+				expectedValue.Token.Amount = expectedValue.Token.Amount.Add(tc.newLiquidity.AmountOf(i))
+				expectedPoolAssetsByDenom[i] = expectedValue
+			}
```

Comment:
This suggestion is twofold:
- we are iterating over a map so "i" feels confusing because this identifier is a convention for counters in a loop, not map keys. Suggesting to change to either `denom` or `key`
- `expectedValue := asset` is a copy by value so we do not need to initialize a new `balancer.PoolAsset{...}` directly, can copy instead to simplify

```suggestion
			for denom, asset := range tc.originalPoolAssets {
				expectedValue := asset
				expectedValue.Token.Amount = expectedValue.Token.Amount.Add(tc.newLiquidity.AmountOf(denom))
				expectedPoolAssetsByDenom[denom] = expectedValue
			}
```

Reply:
Great points